src/python/pants/engine/internals/build_files_test.py

```diff
@@ -417,3 +433,87 @@ def assert_resolve_failure_type(
             expected_type,
             f"type was not {expected_type.__name__}. Instead was {type(failure).__name__}, {failure!r}",
         )
+
+
+class PreludeParsingTest(unittest.TestCase):
+    def test_good_prelude(self) -> None:
+        symbols = run_rule(
+            evalute_preludes,
+            rule_args=[
+                PreludeOption(["prelude"]),
+                BuildFileImportsBehaviorWrapper(BuildFileImportsBehavior.warn),
+            ],
+            mock_gets=[
+                MockGet(
+                    product_type=Snapshot,
+                    subject_type=PathGlobs,
+                    mock=lambda _: Snapshot(Digest("abc", 10), ("/dev/null/prelude",), ()),
+                ),
+                MockGet(
+                    product_type=FilesContent,
+                    subject_type=Digest,
+                    mock=lambda _: FilesContent(
+                        [FileContent(path="/dev/null/prelude", content=b"def foo(): return 1")]
+                    ),
+                ),
+            ],
+        )
+        self.assertEqual(symbols.symbols["foo"](), 1)
```

Comment:
Nit: preferred to use `assert` because Pytest does some magic to show how the values were derived. See https://docs.pytest.org/en/latest/assert.html.